tests/integration/test_unit.py

```diff
@@ -7,6 +7,22 @@
 from .. import base
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_unit_public_address(event_loop):
+    async with base.CleanModel() as model:
+        app = await model.deploy(
+            'cs:ubuntu-0',
+            application_name='ubuntu',
+            series='trusty',
+            channel='stable',
+        )
+
+        for unit in app.units:
```

Comment:
Can we assert we got at least one unit, otherwise we skip the assertions

Reply:
Not sure what you mean, are you saying that getting the public_address of one the units is enough?

Reply:
You need to ensure that units is >= 1, otherwise the test will pass and you won't know it's failed as the for loop looks for units. What happens if units is zero for example?

Reply:
Yeah I'm dumb, you're right I should probably use `num_units=2` or something with the `deploy` and then wait until the units are up, that way it's ensured we'll have some units. 
An assertion about units afterwards certainly wouldn't hurt, though might be redundant, because if we have no units after using `num_units`, then we have much bigger problems, so some other test (prob. about deploy) should already be failing. I'll make the changes :+1: